### ISSUE TITLE
is extract_features_dataloader internal?

### ISSUE THREAD

**Issue by crdsteixeira** [fraunhoferportugal/pymdma#28]
standard dataloader from pytorch returns a `tuple(batch, label)`. 
function `extract_features_dataloader` expects a `tuple(batch, labels, img_ids)`.

is this an internal function? it does not work with most common dataloaders.

**Comment by ivo-facoco** [comment]
Thank you for your question!

The `extract_features_dataloader` function is mainly intended for internal use by the API. 

Our Dataset includes sample IDs to link instance-wise metrics to each sample, which PyTorch's default logic doesn’t support.

You can either:
 - Use the recommended `extract_features_from_files` method.
 - Create your own Dataset that returns `tuple(batch, labels, ids)` to leverage PyTorch's batch loading.

You can find an example Dataset [here](https://github.com/fraunhoferportugal/pymdma/blob/main/src/pymdma/image/data/simple_dataset.py)

**Comment by crdsteixeira** [comment]
Thanks for the clarification and support materials!

In this way, may I suggest that the function is marked as internal (with the single underscore convention)? would make it more explicit.

yes, one can use `extract_features_from_files` but it involves extra steps depending on the specific project implementation structure. I still think [NIT] having a function that extracts from dataloader for outside users would be very helpful for them, and simplify/enhance  the pakage usage, since it could provide a more standardized implementation :)

**Comment by crdsteixeira** [comment]
this is great! thanks for the quick support <3
